### PR TITLE
Add native testing / cleanup GraalVM configuration

### PR DESCRIPTION
**Pull request by graemerocher**
* Fixes GraalVM Native Image support by cleaning up configuration
* Adds native testing
* Cleanups deprecation warnings
* Fixes build warnings

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -2,6 +2,7 @@
 micronaut = "4.0.0-SNAPSHOT"
 micronaut-docs = "2.0.0"
 micronaut-test = "4.0.0-SNAPSHOT"
+micronaut-gradle-plugin = "3.7.5"
 groovy = "4.0.7"
```
Comment (sdelamo) [inline gradle/libs.versions.toml:6]:
```suggestion
groovy = "4.0.9"
```

File: micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/CountedInterceptor.java [hunk 2]
```diff
@@ -49,45 +51,53 @@ public class CountedInterceptor implements MethodInterceptor<Object, Object> {
     public static final String RESULT_TAG = "result";
 
     private final MeterRegistry meterRegistry;
+    private final ConversionService conversionService;
 
     public CountedInterceptor(MeterRegistry meterRegistry) {
```
Comment (sdelamo) [inline micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/CountedInterceptor.java:56]: Should we annotate this with `@Deprecated`?

File: micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java [hunk 3]
```diff
@@ -76,15 +79,27 @@ public class TimedInterceptor implements MethodInterceptor<Object, Object> {
     private static final Logger LOGGER = LoggerFactory.getLogger(TimedInterceptor.class);
 
     private final MeterRegistry meterRegistry;
+    private final ConversionService conversionService;
 
     /**
      * @param meterRegistry The meter registry
      */
     protected TimedInterceptor(MeterRegistry meterRegistry) {
```
Comment (sdelamo) [inline micrometer-core/src/main/java/io/micronaut/configuration/metrics/micrometer/intercept/TimedInterceptor.java:87]: should we annotate this with `@Deprecated` and add the `@deprecated` javadoc

File: test-suite/micronaut-serialization/src/test/java/io/micronaut/configuration/metrics/serialization/MetricsMicronautSerializationTest.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import java.util.Map;
+
+import io.micronaut.http.client.HttpClient;
+import io.micronaut.http.client.annotation.Client;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+@MicronautTest
+class MetricsMicronautSerializationTest {
```
Comment (sdelamo) [inline test-suite/micronaut-serialization/src/test/java/io/micronaut/configuration/metrics/serialization/MetricsMicronautSerializationTest.java:13]: I assume you moved this to junit to be able to test it with native as well?

**Review by sdelamo — APPROVED** [review]

File: test-suite/micronaut-serialization/src/test/java/io/micronaut/configuration/metrics/serialization/MetricsMicronautSerializationTest.java (hunk 4 above)
Comment (graemerocher) [reply test-suite/micronaut-serialization/src/test/java/io/micronaut/configuration/metrics/serialization/MetricsMicronautSerializationTest.java:13]: correct

**Pull request merged**